src/commands/status.js

```diff
@@ -0,0 +1,118 @@
+const Listr = require('listr');
+
+const os = require('os');
+const publicip = require('public-ip');
+const prettyms = require('pretty-ms');
+const prettybyte = require('pretty-bytes');
+
+const BaseCommand = require('../oclif/command/BaseCommand');
+
+const UpdateRendererWithOutput = require('../oclif/renderer/UpdateRendererWithOutput');
+
+const MutedError = require('../oclif/errors/MutedError');
+
+const PRESETS = require('../presets');
+
+class StatusCommand extends BaseCommand {
+  /**
+   * @param {Object} args
+   * @param {Object} flags
+   * @param {DockerCompose} dockerCompose
+   * @return {Promise<void>}
+   */
+  async runWithDependencies(
+    {
+      preset,
+    },
+    flags,
+    dockerCompose,
+  ) {
+    const status = {
+      host: {},
+      docker: {},
+      masternode: {},
+    };
+    const tasks = new Listr([
+      {
+        title: 'Show host status',
+        task: () => (
+          new Listr([
+            {
+              title: 'Fetch hostname',
+              task: () => status.host.hostname = os.hostname()
+            },
+            {
+              title: 'Fetch uptime',
+              task: () => status.host.uptime = prettyms(os.uptime() * 1000)
+            },
+            {
+              title: 'Fetch platform',
+              task: () => status.host.platform = os.platform()
+            },
+            {
+              title: 'Fetch user info',
+              task: () => status.host.user = os.userInfo().username
+            },
+            {
+              title: 'Fetch load average',
+              task: () => status.host.loadavg = os.loadavg()
+            },
+            {
+              // Waiting for feature: https://github.com/nodejs/node/pull/31351
+              title: 'Fetch disk info',
+              enabled: false,
+              task: () => status.host.diskfree = 0
+            },
+            {
+              title: 'Fetch memory info',
+              task: () => status.host.memfree = prettybyte(os.totalmem()) + ' / ' + prettybyte(os.freemem())
+            },
+            {
+              title: 'Fetch CPU info',
+              task: () => status.host.cpus = os.cpus().length
+            },
+            {
+              title: 'Fetch IP info',
+              task: async () => {
+                status.host.ip = await publicip.v4();
+              }
+            },
+          ])
+        )
+      },
+      {
+        title: 'Show Docker status',
+        task: () => (
+          new Listr([
+            {
+              title: 'Show running container states',
```

Comment:
We could add tasks for each service like "Drive is running"  and use isServiceRunning to check status

Reply:
OK, I tried implementing this as a loop on all existing containers to make sure we have visibility on any containers running on the system from other networks or previous attempted installations, since these may help with scenarios like troubleshooting a container stuck in restarting state due to blocked ports. I think using `isServiceRunning` makes the command less useful because we only see the Boolean result of the comparison `(status === 'running')`. This means we lose important details about other possible containers states, e.g. "exited", "restarting", "paused", "dead" etc.

Reply:
I got it. Then I think you can put this info to the table as well and you don't need Listr at all.

Reply:
OK, I have re-implemented output using tables, separated the subcommands out to show status for `(host|docker|masternode)` separately, and added a few examples of how I want to execute commands inside the containers. Is this the right approach, or should I move more logic out into functions? It's probably best if we resolve any remaining code style and output issues and close this PR before I continue adding new output.